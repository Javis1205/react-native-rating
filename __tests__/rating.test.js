import { Easing } from 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Rating from 'rating'

const images = {
  starFilled: require('../assets/star_filled.png'),
  starUnfilled: require('../assets/star_unfilled.png')
}

// eslint-disable-next-line
const logCurrentRating = console.log.bind(console)

describe('<Rating />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Rating
        onChange={logCurrentRating}
        selectedStar={images.starFilled}
        unselectedStar={images.starUnfilled}
      />
    )
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with custom props', () => {
    const tree = renderer.create(
      <Rating
        max={6}
        initial={2}
        onChange={logCurrentRating}
        selectedStar={images.starFilled}
        unselectedStar={images.starUnfilled}
        config={{
          easing: Easing.inOut(Easing.ease),
          duration: 350
        }}
        stagger={80}
        maxScale={1.4}
        starStyle={{
          width: 40,
          height: 40,
          backgroundColor: '#5e23dc'
        }}
        containerStyle={{ flexDirection: 'column' }}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
