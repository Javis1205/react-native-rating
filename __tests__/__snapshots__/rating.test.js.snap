// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> renders correctly with custom props 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5e23dc",
        "height": 40,
        "width": 40,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5e23dc",
        "height": 40,
        "width": 40,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5e23dc",
        "height": 40,
        "width": 40,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5e23dc",
        "height": 40,
        "width": 40,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5e23dc",
        "height": 40,
        "width": 40,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5e23dc",
        "height": 40,
        "width": 40,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<Rating /> renders correctly with default props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 1,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "opacity": 0,
            "resizeMode": "contain",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": null,
          }
        }
      />
    </View>
  </View>
</View>
`;
